.evergreen/config.yml

```diff
@@ -319,13 +318,6 @@ tasks:
     vars:
       compile_env: LIBMONGOCRYPT_EXTRA_CFLAGS="-fsanitize=address"
 
-- name: build-and-test-valgrind
-  commands:
-  - func: "fetch source"
-  - func: "build and test"
-    vars:
-      test_env: VALGRIND="valgrind --leak-check=full --error-exitcode=1"
```

Comment:
As far as I am aware, we do not regular run Valgrind tasks, and I feel that the ASan+LSan instrumentation is sufficient to replace it. If you feel otherwise, I can revive the task, but it will take a bit more complexity since Valgrind would need to be inserted as the test launcher for each CTest test.

Reply:
ASAN+LSan covers almost all cases that Valgrind checks except [uninitialized memory reads](https://github.com/google/sanitizers/wiki/AddressSanitizerComparisonOfMemoryTools).

I do not recall any time that the valgrind task failed without ASAN failing. I am not opposed to removing it.